lib/cereal/postgres/postgres.go

```diff
@@ -722,6 +723,10 @@ func (pg *PostgresBackend) DequeueWorkflow(ctx context.Context, workflowNames []
 			return nil, nil, errors.Wrap(err, "failed to retrieve task result for task complete event")
 		}
 		event.TaskResult = tr
+		if err := deleteTaskResult(ctx, tx, taskResultID); err != nil {
+			cancel()
+			return nil, nil, errors.Wrap(err, "failed to remove task result from the database")
```

Comment:
I guess I'm in the camp of _less is more_, if it's an error, that something has failed is implicit:
```suggestion
			return nil, nil, errors.Wrap(err, "remove task result from the database")
```
**but** you folks seem to be in the other camp. That's cool.